In [4]:
#Importar bibliotecas

import pandas as pd 
import plotly_express

In [9]:
#Se carga el dataset de interés
data = pd.read_csv('vehicles_us.csv')

In [10]:
data.info()
data.sample(5)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 51525 entries, 0 to 51524
Data columns (total 13 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   price         51525 non-null  int64  
 1   model_year    47906 non-null  float64
 2   model         51525 non-null  object 
 3   condition     51525 non-null  object 
 4   cylinders     46265 non-null  float64
 5   fuel          51525 non-null  object 
 6   odometer      43633 non-null  float64
 7   transmission  51525 non-null  object 
 8   type          51525 non-null  object 
 9   paint_color   42258 non-null  object 
 10  is_4wd        25572 non-null  float64
 11  date_posted   51525 non-null  object 
 12  days_listed   51525 non-null  int64  
dtypes: float64(4), int64(2), object(7)
memory usage: 5.1+ MB


price  model_year                 model  condition  cylinders    fuel  \
38498   7900      2006.0          dodge dakota       good        8.0     gas   
33676   1950      1997.0       ford expedition  excellent        8.0     gas   
29873   5000      2000.0         toyota tundra       good        8.0     gas   
29305  48000      2016.0     cadillac escalade  excellent        8.0     gas   
36675      1      2010.0  ford f350 super duty  excellent        8.0  diesel   

       odometer transmission    type paint_color  is_4wd date_posted  \
38498  141200.0    automatic   truck      silver     1.0  2019-01-18   
33676  258876.0    automatic     SUV       white     1.0  2018-09-30   
29873  266000.0    automatic  pickup        grey     1.0  2019-03-04   
29305   43870.0    automatic     SUV       black     1.0  2018-08-02   
36675  139486.0    automatic   truck       black     1.0  2018-11-18   

       days_listed  
38498           26  
33676           18  
29873            8  
29305           47  
36675            9

El dataset presenta información sobre vehículos de segunda mano dando información sobre el modelo del mismo, sus características físicas, la fecha en la que se publicó y el costo.
Se tienen 51,525 entradas y 12 columnas de las cuales hay valores ausentes en 5 columnas y en cada una el valor ausente se sustituirá con:
- Model_year: Se usará el 0 para sustituir los valores ausentes.
- cylinders: Se usará un 0 para sustituir los valores ausentes.
- odometer: Se usará el 0 para sustituir los valores ausentes.
- paint_color: Se usará 'Unknown' para sustituir los valores ausentes.
- is_4wd: Se tienen valores de 1 y NaN, por lo que se sustituirá NaN por 0.
        

In [23]:
data['model_year'] = data['model_year'].fillna(0)
data['cylinders'] = data['cylinders'].fillna(0)
data['odometer'] = data['odometer'].fillna(0)
data['paint_color'] = data['paint_color'].fillna('Unknown')
data['is_4wd'] = data['is_4wd'].fillna(0)

In [24]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 51525 entries, 0 to 51524
Data columns (total 13 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   price         51525 non-null  int64  
 1   model_year    51525 non-null  float64
 2   model         51525 non-null  object 
 3   condition     51525 non-null  object 
 4   cylinders     51525 non-null  float64
 5   fuel          51525 non-null  object 
 6   odometer      51525 non-null  float64
 7   transmission  51525 non-null  object 
 8   type          51525 non-null  object 
 9   paint_color   51525 non-null  object 
 10  is_4wd        51525 non-null  float64
 11  date_posted   51525 non-null  object 
 12  days_listed   51525 non-null  int64  
dtypes: float64(4), int64(2), object(7)
memory usage: 5.1+ MB
